### PR TITLE
Update user ext source

### PR DESCRIPTION
**Pull request by BaranekD**

### CODE REVIEW

File: lib/Auth/Process/UpdateUserExtSource.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<?php
+
```
Comment (pajavyskocil) [inline lib/Auth/Process/UpdateUserExtSource.php]: Missing PHPDoc

File: lib/Auth/Process/UpdateUserExtSource.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		$this->adapter = sspmod_perun_Adapter::getInstance(sspmod_perun_Adapter::RPC);
+	}
+
+	public function process(&$request)
+	{
+		assert('is_array($request)');
+		try {
+			$userExtSource = $this->adapter->getUserExtSource($request['Attributes']['sourceIdPEntityID'][0], $request['Attributes']['sourceIdPEppn'][0]);
+			if (is_null($userExtSource)) {
+				throw new SimpleSAML_Error_Exception("sspmod_perun_Auth_Process_UpdateUserExtSource - userExtSource is null");
```
Comment (pajavyskocil) [inline lib/Auth/Process/UpdateUserExtSource.php]: Please use another Exception message - for example: There is no UserExtSource with ExtSource  ... and Login ...

File: lib/Auth/Process/UpdateUserExtSource.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+			}
+
+			$attributes = $this->adapter->getAttributes($userExtSource['id'], array_keys($this->attrMap));
+
+			if (is_null($attributes)) {
+				throw new SimpleSAML_Error_Exception("getting attributes failed.");
```
Comment (pajavyskocil) [inline lib/Auth/Process/UpdateUserExtSource.php]: Please use the same format of Exception message as in other Exceptions

File: lib/Adapter.php [hunk 4]
```diff
@@ -131,6 +131,31 @@ public static function getInstance($interface) {
...
+	 * @param $userExtSource array ext source
+	 */
+	public abstract function updateUserExtSourceLastAccess($userExtSource);
+
+	/**
+	 * @param $userExtSourceId int userExtSourceId
+	 * @param $attributes array attributes
+	 * @return array attributes
+	 */
+	public abstract function getAttributes($userExtSourceId, $attributes);
```
Comment (pajavyskocil) [inline lib/Adapter.php]: Change method name to getUserExtSourceAttributes()

File: lib/Adapter.php [hunk 5]
```diff
@@ -131,6 +131,31 @@ public static function getInstance($interface) {
...
+
+	/**
+	 * @param $userExtSourceId int userExtSourceId
+	 * @param $attributes array attributes
+	 */
+	public abstract function setAttributes($userExtSourceId, $attributes);
```
Comment (pajavyskocil) [inline lib/Adapter.php]: Change method name to setUserExtSourceAttributes()

**Comment by dBucik** [comment]
Tested manually today, all seems to be working just fine.

**Pull request merged**